Reference/using-ioc.md

```diff
@@ -227,3 +220,57 @@ And here is the merged UnityActivator
 			container.Dispose();
 		}
 	}
+
+## LightInject Example
+
+Install the LightInject, LightInject.Mvc and LightInject.WebApi NuGet packages.
+Same as with other Containers, we need to initialize our service container when the Umbraco application started and then register desired services inside of it.
+
+Now create the following files to configure Unity correctly for an Umbraco site.
+
+	public class LightInjectEvents : ApplicationEventHandler
+	{
+        protected override void ApplicationStarted(UmbracoApplicationBase umbracoApplication, ApplicationContext applicationContext)
+		{
+            var container = new ServiceContainer();
+
+            // Our services registrations here... 
+            // Documentation: https://www.lightinject.net/
+
+            // Registration of the Controllers and WebApi Controllers from the executing assemply
+            container.RegisterControllers(Assembly.GetExecutingAssembly());
+            container.RegisterApiControllers(Assembly.GetExecutingAssembly());
+
+            // Enabling required configuration and settings
+            container.EnableMvc();
+            container.EnablePerWebRequestScope();
+            container.EnableWebApi(GlobalConfiguration.Configuration);
+
+            // Setting up the resolver
+            var resolver = new LightInjectWebApiDependencyResolver(container);
+            GlobalConfiguration.Configuration.DependencyResolver = resolver;
+            DependencyResolver.SetResolver(new LightInjectMvcDependencyResolver(container));
+		}
+	}
+    
+If we're going to call/implement this class outside of the Web application or inside of the package for example, it might be useful (and sometimes required) to also register all the other existing Controllers and API Controllers inside of the whole solution. Especially if we'll be willing to use them or inject them in our own registrations. Below code is scanning through the assemblies within the current domain and registering existing controllers which are implementing IHttpController interface.
+
+        private static void RegisterAssembliesServices(IServiceRegistry container)
+        {
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                var controllerTypes = assembly.GetTypes().Where(t => !t.IsAbstract && typeof(IHttpController).IsAssignableFrom(t));
+                foreach (var controllerType in controllerTypes)
+                {
+                    container.Register(controllerType, new PerRequestLifeTime());
+                }
+            }
+        }
+        
+We can call the helper method after our own services registrations.
+
+## Our.Umbraco.IoC
+
+There is also another way of dealing with IoC containers in Umbraco solution and it's with using the package created by one of the Core developers - Shannon Deminick - called: Our.Umbraco.IoC (https://github.com/Shazwazza/Our.Umbraco.IoC).
```

Comment:
I think this should go at the very top, people don't need to manually implement all of this and it would save a lot of headaches to just use this as the recommended approach.